### ISSUE TITLE
aTV 4 compatibility problems

### ISSUE THREAD

**Issue by robbiet480** [postlund/pyatv#7]
Working:
- `play`
- `select` (when playing this will play/pause)
- `topmenu`
- `menu`
- `next`
- Artwork
- Metadata

Not working:
- `pause` pauses for about 10 ms and then plays again
- `previtem` just goes to the next item in the playlist
- `left`, `right`, `up`, `down` throw `pyatv.exceptions.NotSupportedError`

**Comment by postlund** [comment]
One idea here... I know you mentioned that you issued that command manually, but if you add ``--debug``, is the command issued only once? Just so that atvremote doesn't send it twice in rapid order.

**Comment by robbiet480** [comment]
Even if I use the URL directly in a browser or with cURL I'm getting the issue, so this isn't a pyATV problem. I'm trying to sniff the traffic off my iPhone right now but not having luck with Charles, rvitool, Wireshark or mitmproxy. Any ideas?

**Comment by postlund** [comment]
Ok, good. It can be a bit tricky to sniff depending on the switch you have. Most new switches doesn't send out all packets on a port even when you enter promiscuous mode on it. In my switch, which is configurable with a web interface, I had to configure a specific port to be able to receive all packets like that. Maybe you have to do something similar?

**Comment by robbiet480** [comment]
Updated with more issues found :(

**Comment by postlund** [comment]
These issues seems a bit odd to me. Would be really good if you figured out how to sniff the app.

**Comment by robbiet480** [comment]
Updated with exact working/not working things.

`left` and `right` throw a NotSupportedError

**Comment by postlund** [comment]
Arrow keys are not implemented yet (see here: https://github.com/postlund/pyatv/blob/master/pyatv/internal/apple_tv.py#L90). I do not know how they work on gen 3 devices yet, but for later devices the information here should work: http://stackoverflow.com/questions/35355807/has-anyone-reversed-engineered-the-protocol-used-by-apples-ios-remote-app-for-c

You can create an issue (and add the feature label to it) if you want.

**Comment by postlund** [comment]
Could you maybe try and remove the ``prompt-id`` parameter in pyatv.internal.apple_tv.BaseAppleTV.ctrl_int_cmd and verify if that changes anything?

**Comment by robbiet480** [comment]
Sure

**Comment by robbiet480** [comment]
Oh wait I did try that last night. I was using URLs like `192.168.1.214:3689/ctrl-int/1/pause?session-id=21` and it was having the pause issue.

**Comment by postlund** [comment]
This is so strange. From what I can see, this is how it's supposed to work. Did you try in various apps? Music? Video?

**Comment by robbiet480** [comment]
Music it works fine! Let me test in some other apps.

**Comment by robbiet480** [comment]
I was testing in Plex btw

**Comment by robbiet480** [comment]
Pause and play work in Netflix

**Comment by postlund** [comment]
Ok, so it might be app dependent somehow. It's strange that it works as expected in the Remote app though. Maybe it uses the newer MediaRemoteTV protocol (https://github.com/jeanregisser/mediaremotetv-protocol)? Would be nice to verify that.

One last thing that would be interesting to verify if this works (concept):
```
class BaseAppleTV:
    ....
    def press_play(self):
        return (yield from self.daap.get('play'))
    ...
```
And call that from play(). If you have some time over, it would be very helpful if you could verify that as well.

**Comment by robbiet480** [comment]
Running that gave me `CRITICAL: Authentication error: failed to login: 400`

**Comment by robbiet480** [comment]
Absolutely thinking the weird media controls are an issue in Plex only. [This](https://forums.plex.tv/discussion/191765/fast-forward-and-rewind-problem) forum thread makes me very suspicious as well.

**Comment by postlund** [comment]
Ok, that is good to know and probably not much we can do about that. Can we consider this matter closed then and that everything works? Regarding arrow keys, that's a different matter and should be another issue,

**Comment by robbiet480** [comment]
Yup!

**Comment by postlund** [comment]
Maybe this this should be mentioned in the FAQ in the documentation as well?
